### PR TITLE
feat: misc platform cleanup

### LINKED ISSUE #13403: Misc Platform Cleanup
There are a bunch of very small tweaks we can make to `SwirldsPlatform.java` that will improve code cleanliness.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13403

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/nexus/DefaultLatestCompleteStateNexus.java [hunk 1]
```diff
@@ -43,13 +43,13 @@ public class DefaultLatestCompleteStateNexus implements LatestCompleteStateNexus
...
-     * @param stateConfig the state configuration
-     * @param metrics     the metrics object to update
+     * @param platformContext the platform context
      */
-    public DefaultLatestCompleteStateNexus(@NonNull final StateConfig stateConfig, @NonNull final Metrics metrics) {
-        this.stateConfig = Objects.requireNonNull(stateConfig);
-        Objects.requireNonNull(metrics);
+    public DefaultLatestCompleteStateNexus(@NonNull final PlatformContext platformContext) {
+        this.stateConfig = platformContext.getConfiguration().getConfigData(StateConfig.class);
+        ;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/nexus/DefaultLatestCompleteStateNexus.java:50]: ;

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultStateSignatureCollector.java [hunk 2]
```diff
@@ -70,13 +71,13 @@ public class DefaultStateSignatureCollector implements StateSignatureCollector {
      * Start empty, with no known signed states. A signed state is considered completed when it has signatures from a
      * sufficient threshold of nodes.
      *
-     * @param stateConfig        configuration for state
+     * @param platformContext   the platform context
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultStateSignatureCollector.java:74]: doc alignment

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/nexus/DefaultLatestCompleteStateNexus.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/nexus/DefaultLatestCompleteStateNexus.java:50]: Fixed. Too bad spotless doesn't fix this for us.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultStateSignatureCollector.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultStateSignatureCollector.java:74]: fixed

**Review by alittley — APPROVED** [review]

**Pull request merged**